From 4fe24bd6c0411ce9012b1a40dcb3edc2d9bf8d0d Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 6 Mar 2024 18:27:29 +0100
Subject: [PATCH 536/954] feat(st-ddr): add quasi-dynamic register for
 STM32MP21 & STM32MP23

Add STM32MP21 || STM32MP23 flags with STM32MP25 around qd entry of
struct stm32mp_ddr_reg_desc.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6f46b2d1c5f14daedb313117a8e2978c5be44057 feat(st-ddr): add quasi-dynamic register for STM32MP21 & STM32MP23]
Change-Id: I8370af15a74b52080ae49d3d42c45ddb3be1553d
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/364994
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 drivers/st/ddr/stm32mp_ddr.c     | 4 ++--
 include/drivers/st/stm32mp_ddr.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/st/ddr/stm32mp_ddr.c b/drivers/st/ddr/stm32mp_ddr.c
index a0465dbb50..8ef91f7223 100644
--- a/drivers/st/ddr/stm32mp_ddr.c
+++ b/drivers/st/ddr/stm32mp_ddr.c
@@ -43,7 +43,7 @@ void stm32mp_ddr_set_reg(const struct stm32mp_ddr_priv *priv, enum stm32mp_ddr_r
 			ERROR("invalid parameter offset for %s", desc[i].name);
 			panic();
 		} else {
-#if STM32MP25
+#if STM32MP21 || STM32MP23 || STM32MP25
 			if (desc[i].qd) {
 				stm32mp_ddr_start_sw_done(priv->ctl);
 			}
@@ -51,7 +51,7 @@ void stm32mp_ddr_set_reg(const struct stm32mp_ddr_priv *priv, enum stm32mp_ddr_r
 			value = *((uint32_t *)((uintptr_t)param +
 					       desc[i].par_offset));
 			mmio_write_32(ptr, value);
-#if STM32MP25
+#if STM32MP21 || STM32MP23 || STM32MP25
 			if (desc[i].qd) {
 				stm32mp_ddr_wait_sw_done_ack(priv->ctl);
 			}
diff --git a/include/drivers/st/stm32mp_ddr.h b/include/drivers/st/stm32mp_ddr.h
index a634289ec4..f279f294f2 100644
--- a/include/drivers/st/stm32mp_ddr.h
+++ b/include/drivers/st/stm32mp_ddr.h
@@ -29,7 +29,7 @@ struct stm32mp_ddr_reg_desc {
 	const char *name;
 	uint16_t offset;	/* Offset for base address */
 	uint8_t par_offset;	/* Offset for parameter array */
-#if STM32MP25
+#if STM32MP21 || STM32MP23 || STM32MP25
 	bool qd; /* quasi-dynamic register if true */
 #endif
 };
-- 
2.53.0

